Library/Homebrew/service.rb

```diff
@@ -100,15 +100,31 @@ def error_log_path(path = nil)
       end
     end
 
-    sig { params(value: T.nilable(T::Boolean)).returns(T.nilable(T::Boolean)) }
+    sig { params(value: T.nilable(T.any(T::Boolean, T::Hash))).returns(T.nilable(T::Hash)) }
     def keep_alive(value = nil)
       case T.unsafe(value)
       when nil
         @keep_alive
       when true, false
+        @keep_alive = { always: value }
+      when Hash
```

Comment:
Feels strange to do essentially no validation on this Hash. Additionally, could we consider requiring a specific data type?

Reply:
I'm not sure what the difference is, but I can check if the keys are allowed here.

Reply:
Yeh, I think checking for invalid keys would be good. Is there a reason we could have this be e.g. a `Symbol` or `Hash` always rather than a `Hash` or `Boolean`?